### ISSUE TITLE
Custom Fonts Don't Show On A Real Device

### ISSUE THREAD

**Issue by ShariatPanah** [dotnet/maui#19935]
### Description

Hey, i have added some Fonts to my Resources, and then added in ConfigureFonts, when i run the app on an android emulator (Android 12.0) the fonts show up as expected with no problems. but when i run the app on my real android device (Android 8.0), the fonts won't show up, like they dont exist at all.
so where is the problem? is it a Maui Bug or something that i'm missing?

### Version with bug

8.0.3

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 8.0 and 12.0

**Comment by drasticactions** [comment]
Could you please create an example project showing how you're adding the fonts?

**Comment by ShariatPanah** [comment]
i add the fonts to this directory: Resources\Fonts and I make sure the BuildAction of the fonts is set to MauiFont, then i register these fonts in ConfigureFonts method. this is the process i do.
and here is a repro project: [https://github.com/ShariatPanah/FontsBug/](https://github.com/ShariatPanah/FontsBug/)

**Comment by Zhanglirong-Winnie** [comment]
Verified this issue with Visual Studio 17.10.0 preview2 (8.0.14). Not repro on android device with sample project. 
I can't repro this issue on Xiaomi, Google, and Samsung devices.

**Comment by ShariatPanah** [comment]
@Zhanglirong-Winnie did you test on real devices? could you please tell me what android version are they?

**Comment by Zhanglirong-Winnie** [comment]
> @Zhanglirong-Winnie did you test on real devices? could you please tell me what android version are they?

Yes, I am testing on real devices, I am using XiaoMi (Android 12.5), Samsung (Android 13), Google (Android12).

**Comment by ShariatPanah** [comment]
@Zhanglirong-Winnie I'm testing on a real device Huawei mate 10 lite (Android 8.0 Api 26) and for this Font I provided in the sample (Raleway.ttf) it won’t apply to the text at all, but on an emulator (Pixel 5 Android 12.0 Api 31) it works just fine.

but I opened another issue [here](https://github.com/dotnet/maui/issues/20744) and I described a really strange behavior I’m working with IRANSansMobile(FaNum).ttf on that repro and I can provide the font if you want.
I just post the screenshot:
![306600048-6ae6895b-a2f9-4d86-b903-d1111a6949e6](https://github.com/dotnet/maui/assets/32509267/d827d122-e298-41d7-9cca-aff5397f6302)

so for summary:
the font IRANSansMobile(FaNum).ttf which is a Farsi font the font will apply to just letters (Not to Numbers, meaning the numbers remain on English Font) on my Real Device (Huawei Android 8.0) but on the Emulator it applies to all numbers and letters as expected.
on the other hand for Raleway.ttf it won’t apply to any text on Huawei, but works just fine on the Emulator.

I updated the maui to version 8.0.14 and I don’t see any fix here, at least not for my Device (Huawei)
So I don’t know maybe I’m missing something here?
